src/pages/TutorProfilePage.tsx

```diff
@@ -0,0 +1,792 @@
+import React, { useState, useEffect } from 'react';
+import { Formik, FormikHelpers, Field, FieldProps } from 'formik';
+
+import { tutorValidationSchema } from '../validationSchemas';
+import {
+    Box,
+    Button,
+    SimpleGrid,
+    HStack,
+    Spacer,
+    Grid,
+    Avatar,
+    AvatarBadge,
+    Input,
+    VStack,
+    WrapItem,
+    FormControl,
+    FormErrorMessage,
+    FormLabel,
+    Textarea,
+} from '@chakra-ui/react';
+
+interface TutorProfilePageProps {
+    isUpdate?: boolean;
+}
+import { MultiSelect, Option, useMultiSelect } from 'chakra-multiselect';
+import { AddIcon } from '@chakra-ui/icons';
+import avatar from '../assets/avatar.jpg';
+import { TutorRequest } from '../models/interfaces.ts';
+import axios from 'axios';
+import { theme } from '../util/theme.ts';
+import { headers } from '../util';
+
+const TutorProfilePage: React.FC<TutorProfilePageProps> = ({ isUpdate = false }) => {
+    const { firstName, lastName, email } = JSON.parse(localStorage.getItem('userData') ?? '{}');
```

Comment:
add 1 line 
const [isUpdate, setIsUpdate] = useState(true);

then change isUpdate accorting your logic

Reply:
You suggest to use this state inside the function component, but where is that state will take his value to set? it should come to a component from outside with props.

Reply:
you pass the same props all time. If it have never changed the best way to use in component useState hook and
1) if you get tutor diring fetching data set to true/false (according to your logic)
2) in createTutor function set to true/false  (according to your logic)

Reply:
you pass the same props all time. - not true,
isUpdate prop can be true and false depends on logic from outside the component 
When isUpdate passed in props - means it's true and then update tutor logic works within the component
When not passed - the default value is used : isUpdate=false and then its a Create tutor.